cranelift/codegen/src/isa/aarch64/lower.rs

```diff
@@ -550,7 +550,62 @@ pub(crate) fn lower_address<C: LowerCtx<I = Inst>>(
         return MemArg::RegOffset(reg, offset as i64, elem_ty);
     }
 
-    // Handle two regs and a zero offset, if possible.
+    // Handle two regs and a zero offset with built-in extend, if possible.
+    if addends.len() == 2 && offset == 0 {
+        let parts = if let Some(ext_insn) = maybe_input_insn(ctx, addends[0], Opcode::Uextend) {
```

Comment:
These 4 clauses computing `parts` are a bit duplication-ful.  If the logic gets extended at some point to deal with 16- or 8-bit indices, it'll be even worse.  Hence: is there some way to reduce the level duplication, without making it harder to follow?

Reply:
Sure, factored out (i) the signed/unsigned duality, with a new helper `maybe_input_insn_multi`; and (ii) the two commutative cases, with a for-loop. Thanks!